Fix indexes in core tables (#2247)
microsoft/dicom-server: src/Microsoft.Health.Dicom.SqlServer/indexes.md

# Indexes

This document provides context on how the SQL indexes are organized on core tables.

## Context

Study, Series and Instance table all have PartitionKey, *Keys and *Uid's. 
- `*Uids` are customer generated unique ids. 
- `*Keys` are system generated primary keys. Key's are smaller and used for internal SQL joins only. Should not be distributed to customers.
- `PartitionKey` allows logical grouping of data per partition. Losely held, no strong enforcement. Need to be extremely carefull to include in all filtering and exceptions should be carefully reviewed.

## Best Practices

> - All the indexes will be using PartitionKey as first column, except for columns for QIDO, they will have this column at the end to support cross partition queries in the future. All the SQL filtering should include PartitionKey.

Reviewer: Based on your discussion with @bcarthic, I agree that we should optimize for our known use cases.
```suggestion
- All indexes will use PartitionKey as the first column. All SQL filtering should include PartitionKey.
```
